### PR TITLE
issue 71: reading from java beans

### PR DESCRIPTION
**Pull request by kuba-mikulski**

### CODE REVIEW

**Comment by krzemin** [comment]
Also, please remember about running scalafmt before comitting - otherwise travis build will fail.

File: chimney/src/main/scala/io/scalaland/chimney/internal/MacroUtils.scala [hunk 1]
```diff
@@ -26,9 +26,9 @@ trait MacroUtils extends CompanionUtils {
       }
     }
 
-    def getterMethods: Iterable[MethodSymbol] = {
+    def parameterlessMethods: Iterable[MethodSymbol] = {
       t.decls.collect {
-        case m: MethodSymbol if m.isGetter =>
+        case m: MethodSymbol if (m.paramLists.isEmpty || m.paramLists == List(List())) && !m.isPrivate =>
```
[the remark this answers is not in the document]
Comment (krzemin) [reply chimney/src/main/scala/io/scalaland/chimney/internal/MacroUtils.scala]: Note that this is also changed in #76 in order for one test to pass. Whichever PR comes first, please keep their behavior (can be factored out to a separate method of course).

**Comment by krzemin** [comment]
Travis build failed due to code being not properly formatted. Please run `scalafmt` sbt task and push again.

**Comment by kuba-mikulski** [comment]
Addressed the comments in this PR, but Codacy is now complaining due to too many ifs in `DslBlackboxMacros`, and I don't really see a way of converting it to case matching due to the values being dynamic. Thoughts?

**Comment by krzemin** [comment]
I think you can ignore Codacy - it was complaining for every such if-branch. Instead, I would focus on adding few more test cases to keep the coverage. At least:
- case when the switch is disabled and we don't interpret bean naming conventions
- case when a field named `isXXX` doesn't have Boolean type

**Pull request merged**